### PR TITLE
Start a GUI page

### PR DESCRIPTION
**Pull request by dstansby**
A start on https://github.com/UCL-ARC/python-tooling/issues/354. I don't have recent experience with other GUI toolkits, so leave it for others to fill them in.

I would suggest merging this PR as a starting point, and then iterate on it to add more content in future PRs?

### CODE REVIEW

File: docs/pages/guis.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+# Graphical user interface toolkits
+
+| Name      | Short description                                                                     | 🚦  |
+| --------- | ------------------------------------------------------------------------------------- | :-: |
+| [tkinter] | A Python interface to the Tcl/Tk GUI toolkit. Built into the Python standard library. | 🟠  |
+
+<!-- URLS used above -->
+
+[tkinter]: https://docs.python.org/3/library/tkinter.html
```
Comment (samcunliffe) [inline docs/pages/guis.md]:
```suggestion
| [QtPy]    | A Python interface to the [Qt] GUI system. Recommended if you need to be binding-independent. | 🟢 |
| [PySlide] | A Python binding to the [Qt] GUI system. Recommended if you're developing an app that you'll build and distribute as an executable. In which case you don't need to be binding-independent.  | 🟢 |
| [PyQt] | A Python binding to the [Qt] GUI system. A specific binding but with a more restrictive license.  | 🟠  |
| [tkinter] | A Python interface to the Tcl/Tk GUI toolkit. Built into the Python standard library. | 🟠  |
<!-- URLS used above -->

[tkinter]: https://docs.python.org/3/library/tkinter.html
[Qt]: https://www.qt.io/
[QtPy]: https://github.com/spyder-ide/qtpy
[PySlide]: https://wiki.qt.io/Qt_for_Python
[PyQt]: https://wiki.python.org/moin/PyQt
```

Comment (samcunliffe) [reply docs/pages/guis.md]: Taken from @p-j-smith's comment
- https://github.com/UCL-ARC/python-tooling/issues/354#issuecomment-2051416122

**Comment by samcunliffe** [comment]
I'm still 👎 about a different page for each type of library and think we should have a large(ish) single page with all libraries that we recommend/use and like in our projects.

**Comment by dstansby** [comment]
> I'm still 👎 about a different page for each type of library and think we should have a large(ish) single page with all libraries that we recommend/use and like in our projects.

👍 for this in theory (especially as the list of recommendations grows), but deserves it's own issue/PR?

**Comment by nicolin** [comment]
I have just joined and use Qt C++ for GUI development, so a GUI page should sit outside python tooling as a GUI should be solver code agnostic just providing input to the solver and also reading back as xml or text format.

**Comment by samcunliffe** [comment]
> I have just joined and use Qt C++ for GUI development, so a GUI page should sit outside python tooling as a GUI should be solver code agnostic just providing input to the solver and also reading back as xml or text format.

The scope of this is research software projects in general. We are not just for solvers. I think there is enough Python-specific interest in GUI programming and enough Python GUI libraries that the page should exist for signposting.

**Comment by samcunliffe** [comment]
... although we should do something with this PR. 
Merge as-is or move the table to utilities? I favor the latter but I'd bow to the consensus.

**Comment by dstansby** [comment]
I'm 👍 to as is, as I think having a GUI page makes it more discoverable.

**Comment by paddyroddy** [comment]
> I have just joined and use Qt C++ for GUI development, so a GUI page should sit outside python tooling as a GUI should be solver code agnostic just providing input to the solver and also reading back as xml or text format.

I second Sam. The purpose of this project is not to imply that GUI means only for Python. We're saying in the context of Python development, if you want advice on GUIs then here are some potential ideas, and our experience.

**Comment by p-j-smith** [comment]
> > I'm still 👎 about a different page for each type of library and think we should have a large(ish) single page with all libraries that we recommend/use and like in our projects.
>
> 👍 for this in theory (especially as the list of recommendations grows), but deserves it's own issue/PR?
>
I'm in two minds about this, but probably agree that having a single page for these things would be good. Should the libraries in [Parallel and asychronous processing](https://github-pages.arc.ucl.ac.uk/python-tooling/pages/parallel-async.html) also be moved to this `libraries` page? Agree with @dstansby that having a separate issue / PR to discuss this makes sense

**Comment by samcunliffe** [comment]
Merge as-is then?

**Review by p-j-smith — APPROVED** [review]
looks good - shall we merge this and continue discussion around a single libraries page in https://github.com/UCL-ARC/python-tooling/issues/427?

**Pull request merged**